packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -87,87 +128,112 @@ contract PluginSetupProcessor is DaoAuthorizable {
     error PluginRepoNonexistent();
 
     /// @notice Thrown if a plugin setup is not prepared.
-    error SetupNotPrepared();
+    /// @param setupId The abi encoded hash of versionTag & permissions & helpers.
+    error SetupNotPrepared(bytes32 setupId);
 
     /// @notice Thrown if a plugin setup was already prepared.
-    error SetupAlreadyPrepared();
+    /// @param setupId The abi encoded hash of versionTag & permissions & helpers.
+    error SetupAlreadyPrepared(bytes32 setupId);
 
     /// @notice Thrown if a plugin setup is not applied.
     error SetupNotApplied();
 
     /// @notice Thrown if a plugin setup was already prepared. This is done in case the `PluginSetup` contract is malicios and always/sometime returns the same addresss.
     error SetupAlreadyApplied();
```

Comment:
Shouldn't these errors also return the `setupId`?

Reply:
would the `setupId` give you that information ? you would need to be encoding the same stuff on the SDK to know what `setupId` emitted from contract corresponds to.  

Does it give that advantage ? but adding it really doesn't make anything costly, so I will add it.

Reply:
Haha, I realized these errors are not thrown at all. So, I removed them. https://github.com/aragon/core/pull/164/commits/79f40f9ea098b99cc3d6299406b30a31d757af1d